GTG/core/saved_searches.py

```diff
@@ -44,7 +44,12 @@ def __init__(self, id: UUID, name: str, query: str) -> None:
         self._query = query
         self._icon = None
 
-        super(SavedSearch, self).__init__()
+        super().__init__()
+
+    # TODO probably not needed
```

Comment:
I think we should just remove that and use the constructor in tests. The only thing the method adds is that you don't have to generate the UUID yourself, which I don't think is worth it.

It looks like the code only creates a SavedSearch object when reading from XML. I haven't looked at what happens on the GTK side when we create one, but this concerns me a bit.

Reply:
I have removed it and adapted the test.

> It looks like the code only creates a SavedSearch object when reading from XML. I haven't looked at what happens on the GTK side when we create one, but this concerns me a bit.

I'm not sure what you mean here. Do you mean usages like I mentioned [in the other comment](https://github.com/getting-things-gnome/gtg/pull/1036#discussion_r1517246168)?

Reply:
Oh yeah, I missed that. Please ignore my comment ^^